### ISSUE TITLE
Syntax highlighting

### ISSUE THREAD

**Issue by Leon99** [cake-build/cake-rider#178]
Installing this plugin disables all syntax highlighting in a .cake file. Is that by design?

**Comment by nils-a** [comment]
@Leon99 well, the answer is yes, that is currently by design.

Rider, in it's default, somehow assumes some language for Cake files. (I've not found out which, currently.)
When installing Cake for Rider, we define a language for Cake files (this is needed for most of the things the plugin does) and this definition in it's current state does not include syntax highlighting, sadly.

**Comment by Leon99** [comment]
Sad indeed, editing C# code without syntax highlighting is not nice at all.

**Comment by nils-a** [comment]
Agreed. Sadly "extending" on existing languages is not possible in Rider. So we can't re-use what's available for c# and have to re-implement everything. Currently the only IDE supporting auto-complete and highlighting is the [Cake extension for VS Code](https://cakebuild.net/docs/integrations/editors/vscode/intellisense). 
(There, that's done via OmniSharp which sadly is not an option for Rider)

**Comment by js6pak** [comment]
> So we can't re-use what's available for c# and have to re-implement everything

That completely defeats the point of a rider plugin imo. If you want to use omnisharp, just use vscode.

> Sadly "extending" on existing languages is not possible in Rider

Are you sure? Don't know hacky it will be yet, but I'm pretty sure all we have to do is to just "fake" our own C# project to resharper.

**Comment by nils-a** [comment]
> Are you sure? Don't know hacky it will be yet, but I'm pretty sure all we have to do is to just "fake" our own C# project to resharper.

This is, in essence, what O# does when processing Cake. I've never tried it that way.. You're very welcome to give it a try.

**Comment by nils-a** [comment]
@garywoodfine The best resource I know is the *Custom Language* part of the SDK. You'd best start with [*Implementing a lexer*](https://plugins.jetbrains.com/docs/intellij/implementing-lexer.html) and then continue over the following two chapters.

However, there is also (a bit undocumented, though) the [*Embedded Language*](https://plugins.jetbrains.com/docs/intellij/implementing-lexer.html#embedded-language) feature. I've always wanted to test if that could be utilized to "overlay" the cake features over any existing language.

On a sidenote: Learning Kotlin was very straight forward for me. In part it's very similar to C# and Java and it is simply a brilliant language. (You can use the IntelliJ in the community edition to work on the plugin.)

**Comment by garywoodfine** [comment]
Thanks I will take a look and set myself onto a new learning adventure!!
